### PR TITLE
feat: TT-153 Sign Up Validation Function

### PR DESCRIPTION
**Pull request by israteneda**

### CODE REVIEW

File: SignUpValidation/sample.dat [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "displayName": "Test User",
+  "givenName": "Test User",
+  "identities": [
+    {
+      "signInType": "federated",
+      "issuer": "google.com",
+      "issuerAssignedId": "102373679322388401018"
+    }
+  ]
+}
```
Comment (enriquezrene) [inline SignUpValidation/sample.dat:14]: What is this file intended for?

File: SignUpValidation/function.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (enriquezrene) [inline SignUpValidation/function.json:1]: I don't think this functionality should be implemented as an Azure function. Why not making it part of the API?

File: SignUpValidation/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  // Log the request body
+  context.log(`Request body: ${JSON.stringify(req.body)}`);
+
+  // Get domain of email address
+  const domain = req.body.email.split("@")[1];
+  const allowedDomains = ["ioet.com"];
+
+  context.log("Validation: ", allowedDomains.includes(domain.toLowerCase()));
+  // Check that the domain of the email is from a specific other tenant
+  if (!allowedDomains.includes(domain.toLowerCase())) {
```
Comment (enriquezrene) [inline SignUpValidation/index.js:49]: You're only looking for `ioet.com` as a valid domain?
I don't think so, you should find a way to make sure the token is correct and it is not expired.

File: SignUpValidation/sample.dat (hunk 1 above)
Comment (israteneda) [reply SignUpValidation/sample.dat:14]: [`sample.dat` is primarily for Azure Portal web interface usage](https://github.com/Azure/Azure-Functions/issues/1241)
It's used when we want to try the function on Azure
image

File: SignUpValidation/function.json (hunk 2 above)
Comment (israteneda) [reply SignUpValidation/function.json:1]: Yep is a good idea, I created as azure function because the sign-up validation was part of the azure scope and azure provides some examples with azure function to validate the sign-up process with the `API Connector` tool, but I can create a ticket to migrate this function to the API, because now is implemented in the user flow.

File: SignUpValidation/index.js (hunk 3 above)
Comment (israteneda) [reply SignUpValidation/index.js:49]: At this point we didn't have the token yet.
Since is before the user is created, and it's validating the email with which the user is trying to log in with the google provider.

**Review by Angeluz-07 — APPROVED** [review]

**Review by diegotony — APPROVED** [review]
:+1:
